chromium_src/components/sync_device_info/device_info_sync_bridge_unittest.cc

```diff
@@ -218,5 +218,45 @@ TEST_F(DeviceInfoSyncBridgeTest, BraveExpireOldEntriesUponStartup) {
                   Pair(specifics_old.cache_guid(), _)));
 }
 
+TEST_F(DeviceInfoSyncBridgeTest, BraveResetsProgressMarkerOnce) {
+  const DeviceInfoSpecifics specifics = CreateLocalDeviceSpecifics();
+  ModelTypeState model_type_state = StateWithEncryption("ekn");
+  model_type_state.mutable_progress_marker()->set_token("ABC");
+  WriteToStoreWithMetadata({specifics}, model_type_state);
+
+  base::RunLoop run_loop;
```

Comment:
normally you should have two base::RunLoop for each scope with a pair of Run and Quit.
Your second RunLoop::Quit does nothing because you get lucky that `RestartBridge()` call `PumpAndShutdown()` which contains a `base::RunLoop().RunUntilIdle`

Reply:
Thanks @darkdh , I see now the problems.

Fixed that at https://github.com/brave/brave-core/pull/12005/commits/5e606ddf34cef97fe6298beb90538cddcfa8f3f3 .